qiskit/opflow/gradients/circuit_gradients/lin_comb.py

```diff
@@ -599,7 +623,12 @@ def _gradient_states(
         qr_superpos = QuantumRegister(1)
         state_qc = QuantumCircuit(*state_op.primitive.qregs, qr_superpos)
         state_qc.h(qr_superpos)
-        state_qc.compose(state_op.primitive, inplace=True)
+        # unroll separately from the H gate since we need the H gate to be the first
+        # operation in the data attributes of the circuit
+        unrolled = CircuitGradient._transpile_to_supported_operations(
```

Comment:
Can you just do since this is a subclass of `CircuitGradient`?:

```suggestion
        unrolled = self._transpile_to_supported_operations(
```

Reply:
Sure, thanks for noticing this.